Error in cv::cuda::meanShiftSegmentation ()

The error print is:
modules/core/src/matrix.cpp:1232: error: (-213) You should explicitly call download method for cuda::GpuMat object in function getMat_

```
image = cv::imread(imagePathes[cpt], CV_LOAD_IMAGE_UNCHANGED);
cv::cvtColor(image, image, CV_GRAY2RGBA);
cuImage.upload(image);
cv::cuda::meanShiftSegmentation(cuImage, cuSegmented, 20, 45, 3); // crashing here
cuSegmented.download(segmented);
```

This error could related to [that one](https://github.com/Itseez/opencv/issues/6231)

--- a/modules/cudaimgproc/include/opencv2/cudaimgproc.hpp
+++ b/modules/cudaimgproc/include/opencv2/cudaimgproc.hpp
@@ -639,7 +639,7 @@ CV_EXPORTS void meanShiftProc(InputArray src, OutputArray dstr, OutputArray dsts
 /** @brief Performs a mean-shift segmentation of the source image and eliminates small segments.
 
 @param src Source image. Only CV_8UC4 images are supported for now.
-@param dst Segmented image with the same size and type as src (host memory).
+@param dst Segmented image with the same size and type as src (host or gpu memory).
 @param sp Spatial window radius.
 @param sr Color window radius.
 @param minsize Minimum segment size. Smaller segments are merged.

--- a/modules/cudaimgproc/src/mssegmentation.cpp
+++ b/modules/cudaimgproc/src/mssegmentation.cpp
@@ -372,8 +372,7 @@ void cv::cuda::meanShiftSegmentation(InputArray _src, OutputArray _dst, int sp, 
     }
 
     // Create final image, color of each segment is the average color of its pixels
-    _dst.create(src.size(), src.type());
-    Mat dst = _dst.getMat();
+    Mat dst(src.size(), src.type());
 
     for (int y = 0; y < nrows; ++y)
     {
@@ -389,6 +388,7 @@ void cv::cuda::meanShiftSegmentation(InputArray _src, OutputArray _dst, int sp, 
             dstcol[3] = 255;
         }
     }
+    dst.copyTo(_dst);
 }
 
 #endif // #if !defined (HAVE_CUDA) || defined (CUDA_DISABLER)

